src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,131 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
+import { FormDataKeys } from '../../types/FormatDataKeys';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [formData, setFormData] = useState({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
```

Comment:
```suggestion
  const [newMovie, setNewMovie] = useState(defaultMovie);
```
You can create an object outside the function and reuse it here and when you are resetting the form